### PR TITLE
Fix the failure of lint and accessing tinylicious log in Test Stability pipeline

### PR DESCRIPTION
**Pull request by clarenceli-msft**
[AB#4778](https://dev.azure.com/fluidframework/internal/_workitems/edit/4778)

This PR aims to resolve the recent failures encountered in the Test Stability pipelines. The key changes included are as follows:

1. Sync the `taskLint` section with that in CI build pipeline YAML file

2. Validate the existence of tinylicious log file before proceeding to publish it as an artifact 

The updates were tested on my personal [test branch](https://github.com/microsoft/FluidFramework/tree/test/clarenceli) with reducing the count of test stages. 

The test pipeline run: https://dev.azure.com/fluidframework/internal/_build/results?buildId=171253&view=results

The run after I broke an e2e test intentionally, where 3 different artifacts should exist: https://dev.azure.com/fluidframework/internal/_build/results?buildId=171246&view=results

### CODE REVIEW

File: tools/pipelines/templates/include-test-stability.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ parameters:
   type: boolean
   default: true
 
+- name: taskLintName
```
Comment (tyler-cai-microsoft) [inline tools/pipelines/templates/include-test-stability.yml:18]: I don't see you using this parameter, am I missing something?

**Comment by scarlettjlee** [comment]
Is this partly a problem with the yaml files getting out of sync? If so, can you open a task to refactor them so the parts are shared rather than duplicated?

File: tools/pipelines/test-stability.yml [hunk 2]
```diff
@@ -162,10 +152,23 @@ stages:
...
           poolBuild: ${{ parameters.poolBuild }}
           checkoutSubmodules: ${{ parameters.checkoutSubmodules }}
           timeoutInMinutes: 360
           taskBuild : ${{ parameters.taskBuild }}
-          taskTest: ${{ parameters.taskTest }}
+          taskLint: false
+          taskTest: 
+          - ci:test:mocha
+          - ci:test:jest
+          - ${{ if eq(variables['System.TeamProject'], 'public') }}:
```
Comment (scarlettjlee) [inline tools/pipelines/test-stability.yml:166]: I don't think we would ever run this in the public project.

File: tools/pipelines/templates/include-test-stability.yml (hunk 1 above)
Comment (clarenceli-msft) [reply tools/pipelines/templates/include-test-stability.yml:18]: I missed that.

**Comment by clarenceli-msft** [comment]
> Is this partly a problem with the yaml files getting out of sync? If so, can you open a task to refactor them so the parts are shared rather than duplicated?

I can open a task to refactor them, most work in Install/Build/Lint/Test is duplicated.

File: tools/pipelines/templates/include-test-stability.yml [hunk 3]
```diff
@@ -81,19 +94,18 @@ jobs:
...
 
-    - template: include-install-pnpm.yml
-      parameters:
-        buildDirectory: ${{ parameters.buildDirectory }}
-        # This pipeline experiences a high failure rate when the cache is enabled, so we disable it
-        enableCache: false
+    - ${{ if eq(parameters.packageManager, 'pnpm') }}:
+          - template: include-install-pnpm.yml
+            parameters:
+              buildDirectory: ${{ parameters.buildDirectory }}
```
Comment (tyler-cai-microsoft) [inline tools/pipelines/templates/include-test-stability.yml:100]: Why was this change needed?

File: tools/pipelines/templates/include-test-stability.yml [hunk 4]
```diff
@@ -139,27 +151,25 @@ jobs:
...
           displayName: Publish Artifact - Tinylicious Log
           inputs:
-            targetPath: '${{ parameters.buildDirectory }}/packages/test/test-end-to-end-tests/tinylicious_${{ parameters.stageName }}.log'
+            targetPath: '${{ parameters.buildDirectory }}/packages/test/test-end-to-end-tests/tinylicious.log'
             artifactName: 'tinyliciousLog_${{ parameters.stageName }}'
             publishLocation: 'pipeline'
-          condition: and(failed(), contains('${{ taskTestStep }}', 'tinylicious'))
-          continueOnError: true # Keep running subsequent tasks even if this one fails (e.g. the tinylicious log wasn't there)
+          condition: and(failed(), eq(variables['logExists'], 'true'), contains('${{ taskTestStep }}', 'tinylicious'))
+          continueOnError: true
```
Comment (tyler-cai-microsoft) [inline tools/pipelines/templates/include-test-stability.yml:172]: Why do we need to add this other condition? I'm just curious, non-blocking

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM

**Review by scarlettjlee — APPROVED** [review]

File: tools/pipelines/templates/include-test-stability.yml (hunk 4 above)
Comment (clarenceli-msft) [reply tools/pipelines/templates/include-test-stability.yml:172]: It is to ensure the artifact is published only when the test fails and the tinylicious log is generated. The tinylicious log can be generated even the test passes, however, there exist cases (like stress tests), where the tinylicious log may not be generated even if the test fails. So it is necessary to verify both conditions.

File: tools/pipelines/templates/include-test-stability.yml (hunk 3 above)
Comment (clarenceli-msft) [reply tools/pipelines/templates/include-test-stability.yml:100]: hmmm this change is solely for the sync with CI build pipeline and does not bring any difference compared with the previous version. I have already created a task to refactor Test Stability and CI build pipeline YAML, aiming to unify the duplicate sections by utilizing the same template. It will reduce the overhead associated with addressing issues caused by async
